### ISSUE TITLE
Enhance IELTS Prediction: Refining Data and Models to Boost R² from 0.918990 to 0.999671

### ISSUE THREAD

**Issue by DevManpreet5** [abhisheks008/ML-Crate#663]
### ML-Crate Repository (Proposing new issue)

:red_circle: **Project Title** : Improving Accuracy of Already Existing IELTS Success Analysis and Prediction

:red_circle: **Aim** : To justify and improve the accuracy of the existing IELTS success prediction model in the ML-Crate repository through hyperparameter tuning and the application of various advanced algorithms. The goal is to enhance the R² score from the current value of 0.918990 to 0.999671.

:red_circle: **Dataset** : [IELTS Success Stories Dataset](https://www.kaggle.com/datasets/zakirkhanaleemi/ielts-success-stories-dataset)

:red_circle: **Approach** : The approach will involve thorough data cleaning, exploratory data analysis (EDA), and the implementation of the following machine learning models:
  - GradientBoosting Regression with tuning
  - XGBoost Regression with tuning
  - Random Forest Regression with tuning
  - XGBoost Regression without tuning
  
  I will compare the performance of these models to determine the best algorithm for predicting IELTS success, with the primary metric being the R² score.

**********************************************************************************
### 📍 **Follow the Guidelines to Contribute in the Project :**
* You need to create a separate folder named as the Project Title.
* Inside that folder, there will be four main components.
   * Images - To store the required images.
   * Dataset - To store the dataset or, information/source about the dataset.
   * Model - To store the machine learning model you've created using the dataset.
   * `requirements.txt` - This file will contain the required packages/libraries to run the project in other machines.
* Inside the `Model` folder, the `README.md` file must be filled up properly, with proper visualizations and conclusions.

***********************************************************************
:red_circle::yellow_circle: **Points to Note :**

- The issues will be assigned on a first come first serve basis, 1 Issue == 1 PR.
- "Issue Title" and "PR Title should be the same. Include issue number along with it.
- Follow Contributing Guidelines & Code of Conduct before start Contributing.

***********************************************************************
:white_check_mark: **To be Mentioned while taking the issue :**
- Full name : Manpreet Singh
- GitHub Profile Link : [devmanpreet5](https://github.com/DevManpreet5)
- Participant ID (If not, then put NA) : NA
- Approach for this Project : I will use data cleaning, exploratory data analysis (EDA), and implement the following models: GradientBoosting Regression with tuning, XGBoost Regression with tuning, Random Forest Regression with tuning, and XGBoost Regression without tuning. I will compare the performance of these models to determine the best algorithm for predicting IELTS success. I will also ensure thorough exploratory data analysis before creating any model.
- What is your participant role? (Mention the Open Source Program name. Eg. HRSoC, GSSoC, GSOC etc.) : VSOC 2024

*************************************************************
Happy Contributing 🚀 

All the best. Enjoy your open source journey ahead. 😎

**Comment by DevManpreet5** [comment]
i have all the files ready for same , please assign the same to me . @DevManpreet5
